SerialPrograms/Source/PokemonSV/Programs/PokemonSV_Terarium.cpp

```diff
@@ -0,0 +1,371 @@
+
+                try {
+                    open_map_from_overworld(info, console, context);
+                }
+                catch (...) {
+                    jump_off_wall_until_map_open(info, console, context);
+                }
+
+                //Move cursor to top left corner - even works when at Entrance fly point
+                pbf_press_button(context, BUTTON_ZL, 40, 100);
+                pbf_move_left_joystick(context, 0, 0, 500, 40);
+
+                //Now move toward center
+                if (laggy) {
+                    pbf_move_left_joystick(context, 255, 255, 300, 40); //overshoot by a bit (still works even if not laggy)
+                }
+                else {
+                    pbf_move_left_joystick(context, 255, 255, 250, 40); //250 is more accurate but 300 helps with lag
+                }
+                pbf_press_button(context, BUTTON_ZR, 40, 100);
+
+                try {
+                    //The only pokecenter on the map is Central Plaza
+                    fly_to_closest_pokecenter_on_map(info, console, context);
+                    context.wait_for_all_requests();
+                    returned_to_pokecenter = true;
+                }
+                catch(...) {
+                    console.log("Failed to return to Pokecenter. Closing map and retrying.");
+                    laggy = true;
+                }
+                context.wait_for_all_requests();
+            },
+            {
+                static_cast<VisualInferenceCallback&>(encounter_watcher),
+                static_cast<AudioInferenceCallback&>(encounter_watcher),
+            }
+        );
+        if (ret == 0) {
+            console.log("Battle menu detected.");
+            encounter_watcher.throw_if_no_sound();
+
+            bool is_shiny = (bool)encounter_watcher.shiny_screenshot();
+            if (is_shiny) {
+                console.log("Shiny detected!");
+                pbf_press_button(context, BUTTON_CAPTURE, 2 * TICKS_PER_SECOND, 5 * TICKS_PER_SECOND);
+                throw ProgramFinishedException();
+            }
+            else {
+                console.log("Detected battle. Running from battle.");
+                try{
+                    //Smoke Ball or Flying type required due to Arena Trap
+                    NormalBattleMenuWatcher battle_menu(COLOR_YELLOW);
+                    battle_menu.move_to_slot(console, context, 3);
+                    pbf_press_button(context, BUTTON_A, 10, 50);
+                }catch (...){
+                    console.log("Unable to flee.");
+                    throw OperationFailedException(
+                        ErrorReport::SEND_ERROR_REPORT, console,
+                        "Unable to flee!",
+                        true
+                    );
+                }
+            }
+        }
+    }
+    context.wait_for_all_requests();
+}
+
+void central_to_polar_rest(const ProgramInfo& info, ConsoleHandle& console, BotBaseContext& context) {
```

Comment:
A lot of code duplication here with all the different fly locations.

Since they differ only by the joystick move and the exception string, you move everything into a helper function that takes just the joystick command and the location name for the exception string.